### PR TITLE
Refactor Android uploaders to share more code

### PR DESCRIPTION
**Pull request by badboy**
I moved shared functionality into free-standing functions

We could keep it in the companion object, but it felt a bit weird to re-use functions in another class across its companion object.
Also the one in the derived class just shadows the one from the superclass, so we need to use explicit calls.

### CODE REVIEW

**Comment by badboy** [comment]
Took it to a test run and also ran the UI tests (`./gradlew :glean-sample-app:connectedAndroidTest -PdisablePreDexg`).
Turns out: it was buggy.

Please re-review the latest 2 commits which 1) make it work, 2) make the tests pass

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt [hunk 1]
```diff
@@ -22,6 +22,105 @@ import java.io.FileNotFoundException
...
+private const val LOG_TAG = "glean/PingUploadWorker"
+
+/**
+ * Build the constraints around which the worker can be run, such as whether network
+ * connectivity is required.
+ *
+ * @return [Constraints] object containing the required work constraints
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+fun buildConstraints(): Constraints = Constraints.Builder()
```
Comment (travis79) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt]: This needs marked as `internal`

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt [hunk 2]
```diff
@@ -22,6 +22,105 @@ import java.io.FileNotFoundException
...
+            } else {
+                // Delete files that don't match the UUID FILE_PATTERN regex
+                Log.d(LOG_TAG, "Pattern mismatch. Deleting ${file.name}")
+                file.delete()
+            }
+        }
+    }
+
+    return success
+}
```
Comment (travis79) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt]: I'm not _super_ excited about global stuff shared between multiple classes floating around the top of the PingUploadWorker.kt file.  Would this be better to have in a Util file?

Comment (badboy) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt]: It's not more or less global than it was before as static functions inside a companion object.

I don't like introducing "Util" files as a catch-all for non-class functions.

Comment (travis79) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt]: Okay, not a strong opinion on this, it was just a nit.

**Pull request merged**